hack/lib/image_lib.sh

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+
+source hack/lib/common.sh
+
+semver_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)$"
+
+# docker_login <image_name>
+#
+# docker_login performs a docker login for the server of the provided
+# image if the DOCKER_USERNAME and DOCKER_PASSWORD environment variables
+# are set.
+#
+function docker_login() {
+  if [[ -n "$DOCKER_USERNAME" && -n "$DOCKER_PASSWORD" ]]; then
+    server=$(docker_server_for_image $image_name)
```

Comment:
I ran into a bug with this function being used when pushing the manifest list somewhere other than docker hub.

When we're pushing the images, `$image_name` must be getting set globally in other functions, so it happens to work. When we're pushing the manifest list, `$image_name` never gets set, so this fails for non-docker hub repos. We need to add this line at the beginning of the function, and then update all calls to `docker_login` to pass the `$push_image` as an argument:
```go
image_name=$1; shift || fatal "${FUNCNAME} usage error"
```

Reply:
yes, i found that as  well and have a fix for it. i was going to mention it and see if you wanted a seperate pr. it's tiny.

Reply:
Fixing here is fine I think since it only manifests itself in the manifest push.